brave/brave-ios
# Fix #6606: added UI to handle tx different statuses

## Summary of Changes
For each transaction user clicks confirm in the transaction confirmation screen, it will always have UI to handle the changing of the states. It includes when tx is in loading/submitting, tx has been submitted or signed, tx has been confirmed/completed. 


This pull request fixes #6606 #6861

## Submitter Checklist:


## Test Plan:

## Test regular transaction status changes:
1. Make either a send a swap transaction via Brave Wallet
2. A transaction confirmation screen will pop up
3. click confirm
4. observe a loading screen will show up 
5. observe after the loading, view will become transaction submitted state which has icon, text and buttons
6. click the ok button 
7. observe it will dismiss the whole screen
8. repeat step 1 to 5
9. click the explorer button
10. observe it will open transaction url 
11. repeat step 1 to 5
12. stay at this screen for few seconds
13. observe the view will change to another state indicates the tx has been completed 
14. click the receipt button
15. observe a transaction detail screen show up
16. close the detail screen and click close button
17. observe the whole screen will be dismissed
18. repeat step 1 to 2
19. click advanced setting in the confirmation screen and edit nonce to be `0`. save it
20. click confirm 
21. observe a loading screen will show up
22. observe after loading, view will become transaction failed with an error message
23. click ok button
24. observe it will dismiss the whole screen.

## Test signed transaction:
1. Setup https://github.com/bbondy/eth-manual-tests
2. Change to any test network and visit http://localhost:8081/request.html
3. setup safara debug console to check any output from http://localhost:8081/request.html
4. Click ethereum.enable and grant permission
5. Fill to address next to eth_signTransaction and click it
6. observe a notification will show up and click the notification to open brave wallet
7. once the wallet is unlocked, a send pending transaction will show up 
8. click confirm 
9. observe a loading state will apear
10. observe after loading, the tx status will change to signed. 
11. click ok to close the screen 
12. check account activities to see the transaction appears with status `Signed`
13. Paste the return signed transaction in console to eth_sendRawTransaction and click
14. Check transactions in account activity and previously `Signed` transaction should be `Confirmed` after a while

## Screenshots:


https://user-images.githubusercontent.com/1187676/219449556-8b7b58e1-d907-4dcb-86f5-782230f484fa.mp4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
